src/ln/channel.rs

```diff
@@ -239,18 +240,24 @@ enum ChannelState {
 }
 const BOTH_SIDES_SHUTDOWN_MASK: u32 = (ChannelState::LocalShutdownSent as u32 | ChannelState::RemoteShutdownSent as u32);
 
+
+
+
 // TODO: We should refactor this to be an Inbound/OutboundChannel until initial setup handshaking
 // has been completed, and then turn into a Channel to get compiler-time enforcement of things like
 // calling channel_id() before we're set up or things like get_outbound_funding_signed on an
 // inbound channel.
 pub struct Channel {
+	///TODO: split limits and runtime setting here
+	/// channel limits cant change during channel existence
+	/// but other settings can, but they need an update broadcast before they can change. 
```

Comment:
We don't currently have any settings that are in Channel that we'd ever really want to change at runtime - things like fee requirements and the like are all kept in ChannelManager. I'm still skeptical that we want to store the extra data in each Channel, much easier to pass in a reference to a config object on new_from_req and accept_channel.

The exception, I would say, would be to plumb the limits through to ChannelManager::create_channel and separate out a config that is the config for inbound channels, but then it'd be really nice to not waste the extra space in every Channel object for a config option still, so could be stored somewhere else.